src/components/utils/popover/popover-mobile.ts

```diff
@@ -0,0 +1,146 @@
+import { Popover as PopoverBase } from './popover-abstract';
+import ScrollLocker from '../scroll-locker';
+import { PopoverHeader } from './components/popover-header';
+import { PopoverStatesHistory } from './utils/popover-states-history';
+import { PopoverParams } from './popover.typings';
+import { PopoverItem } from './components/popover-item';
+import { PopoverItem as PopoverItemParams } from '../../../../types';
+import { css } from './popover.const';
+import Dom from '../../dom';
+
+/**
+ * Class responsible for rendering popover and handling its behaviour on mobile screens
+ */
+export class PopoverMobile extends PopoverBase {
+  /**
+   * ScrollLocker instance
+   */
+  private scrollLocker = new ScrollLocker();
+
+  /**
+   * Reference to popover header if exists
+   */
+  private header: PopoverHeader | undefined | null;
+
+  /**
+   * History of popover states for back navigation.
+   * Is used for mobile version of popover,
+   * where we can not display nested popover of the screen and
+   * have to render nested items in the same popover switching to new state
+   */
+  private history = new PopoverStatesHistory();
+
+  /**
+   * Construct the instance
+   *
+   * @param params - popover params
+   */
+  constructor(params: PopoverParams) {
+    super(params);
+
+    /* Save state to history for proper navigation between nested and parent popovers */
+    this.history.push({ items: params.items });
+  }
+
+  /**
+   * Open popover
+   */
+  public show(): void {
+    this.nodes.overlay?.classList.remove(css.overlayHidden);
+
+    super.show();
+
+    this.scrollLocker.lock();
+  }
+
+  /**
+   * Closes popover
+   */
+  public hide(): void {
+    super.hide();
+
+    this.scrollLocker.unlock();
+  }
+
+  /**
+   * Clears memory
+   */
+  public destroy(): void {
+    super.destroy();
+
+    this.scrollLocker.unlock();
+  }
```

Comment:
do we need to clear a state history on `hide` and `destroy`?

Reply:
for now we don't – block tunes recreate popover on each open. In future, when we will need to integrate nested popover to toolbox, I will add clearing history

Reply:
It would be better to add it now since it is a part of memory clearing of the new class